File: TS/58. Length of Last Word.ts
function lengthOfLastWord(s: string): number {
  let count: number = 0;
  let isFirstWordVisited: boolean = false;

  for (let i = s.length - 1; i >= 0; i--) {
    // We meet the last word and we return the count
    if (s[i] == " " && isFirstWordVisited) {
      return count;
    }

    // We meet the first space and we continue
    if (s[i] == " " && !isFirstWordVisited) {
      continue;
    }

    // we meet the first word and we set the flag to true
    if (!isFirstWordVisited) {
      isFirstWordVisited = true;
    }

    count++;
  }

  return count;
}

console.log(lengthOfLastWord("Hello World"));
console.log(lengthOfLastWord("   fly me   to   the moon  "));
console.log(lengthOfLastWord("luffy is still joyboy"));

File: Python/58. Length of Last Word.py
from typing import List

class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        count = 0
        isFirstWordVisited = False 

        for i in range(len(s)-1,-1,-1):
            if s[i] == " " and isFirstWordVisited:
                return count
            
            if s[i] == " " and not isFirstWordVisited:
                continue

            if not isFirstWordVisited:
                isFirstWordVisited = True

            count += 1

        return count

   

if __name__ == "__main__":
    solution = Solution()
    print(solution.lengthOfLastWord("Hello World"))
    print(solution.lengthOfLastWord("   fly me   to   the moon  "))
    print(solution.lengthOfLastWord("luffy is still joyboy"))